packages/checkup-plugin-javascript/src/tasks/eslint-disable-task.ts

```
@@ -63,61 +63,66 @@ export default class EslintDisableTask extends BaseTask implements Task {
+          endLine: node.loc.end.line,
+          endColumn: node.loc.end.column,
+        });
+      }
 
-    get visitors(): Visitor<any> {
-      let self = this;
+      get visitors(): Visitor<any> {
+        let self = this;
 
-      return {
-        visitComment: function (path: any) {
-          if (ESLINT_DISABLE_REGEX.test(path.value.value.trim())) {
-            self.add(path.value);
-          }
+        return {
+          visitComment: function (path: any) {
+            if (ESLINT_DISABLE_REGEX.test(path.value.value.trim())) {
+              self.add(path.value);
+            }
 
-          this.traverse(path);
-        },
-      };
+            this.traverse(path);
+          },
+        };
+      }
     }
-  }
 
-  await Promise.all(
-    filePaths.map((filePath) => {
-      return fs.promises.readFile(filePath, 'utf8').then((fileContents: string) => {
-        let accumulator = new ESLintDisableAccumulator(filePath);
-        let analyzer = new AstAnalyzer<t.File, Visitor<any>, typeof parse, typeof visit>(
-          fileContents,
-          parse,
-          visit,
-          {
-            parser: require('recast/parsers/babel'),
+    await Promise.all(
+      filePaths.map((filePath) => {
+        return fs.promises.readFile(filePath, 'utf8').then((fileContents: string) => {
+          let accumulator = new ESLintDisableAccumulator(filePath);
+          let analyzer;
+          try {
+            analyzer = new AstAnalyzer<t.File, Visitor<any>, typeof parse, typeof visit>(
+              fileContents,
+              parse,
+              visit,
+              {
+                parser: require('recast/parsers/babel'),
+              }
+            );
+            analyzer.analyze(accumulator.visitors);
+            data.push(...accumulator.data);
+          } catch (error) {
+            (error as Error).message = `Error occurred at ${filePath}. ${(error as Error).message}`;
+            this.addNonFatalError(error as Error);
```

I'd probably just throw a new error instead, rather than mutating the current error's message.

---

if i do that, i would lose the stack trace, no?

---

This is true, and since we're wanting to preserve that stack to ultimately output it in the SARIF log, we should keep it this way.